api/oc_network_events.c

```diff
@@ -204,3 +218,17 @@ oc_network_interface_event(oc_interface_event_t event)
   _oc_signal_event_loop();
 }
 #endif /* OC_NETWORK_MONITOR */
+
+int
+oc_get_network_events_queue_length(size_t device)
+{
+  oc_network_event_handler_mutex_lock();
+  int message_count = 0;
+  for (oc_message_t *message = (oc_message_t *)oc_list_head(g_network_events); message != NULL; message = message->next) {
+    if (message->endpoint.device == device) {
+      ++message_count;
+    }
+  }
```

Comment:
move this code to  `static size_t oc_get_network_events_queue_length_locked(size_t device)`
and call the fuction in this place.

Reply:
Redesigned the method to iterate through events passed in param ( non-blocking) 
`static size_t get_events_queue_length(size_t device, oc_list_t events)`